pyatlan/client/asset.py

```diff
@@ -1256,6 +1258,40 @@ def find_term_by_name(
             attributes=attributes,
         )
 
+    def get_hierarchy(
+        self, glossary: AtlasGlossary, attributes: Optional[list[AtlanField]] = None
+    ) -> CategoryHierarchy:
+        from pyatlan.model.fluent_search import FluentSearch
+
+        if attributes is None:
+            attributes = []
+        top_categories: set[str] = set()
+        category_dict: dict[str, AtlasGlossaryCategory] = {}
+        search = (
+            FluentSearch.select()
+            .where(AtlasGlossaryCategory.ANCHOR.eq(glossary.qualified_name or ""))
+            .include_on_results(AtlasGlossaryCategory.PARENT_CATEGORY)
+            .page_size(20)
+            .sort(AtlasGlossaryCategory.NAME.order(SortOrder.ASCENDING))
+        )
+        for field in attributes:
+            search.include_on_results(field)
+        request = search.to_request()
+        response = self.search(request)
+        for category in filter(
+            lambda a: isinstance(a, AtlasGlossaryCategory), response
```

Comment:
Better that we explicitly search for only `AtlasGlossaryCategory` type in the FluentSearch, to push-down the filtering into the search itself

Reply:
@cmgrote Ok but in the JDK you have the following:
` GlossaryCategory.select(client)
                .where(GlossaryCategory.ANCHOR.eq(getQualifiedName()))
                .includeOnResults(GlossaryCategory.PARENT_CATEGORY)
                .includesOnResults(attributes == null ? Collections.emptyList() : attributes)
                .pageSize(20)
                .sort(GlossaryCategory.NAME.order(SortOrder.Asc))
                .stream()
                .filter(a -> a instanceof GlossaryCategory)`
Am I missing something?  I thought this meant the filtering was occurring on the stream and not the search so that the search would retrieve all types of assets and they would be filtered in the stream. So doesn't an equivalent change need to be done there?

Reply:
done

Reply:
Yeah, I should probably drop that line from the Java SDK. The hope with the filter there was more that it would make the explicit cast to a term that comes in the `forEach` redundant, but naturally it's still required in Java (not so in Kotlin).